core/trino-main/src/main/java/io/trino/server/Server.java

```diff
@@ -103,6 +103,7 @@ private void doStart(String trinoVersion)
                 new ServerSecurityModule(),
                 new AccessControlModule(),
                 new EventListenerModule(),
+                new CoordinatorDiscoveryModule(),
```

Comment:
it would be more natural to install this from CoordinatorModule

Reply:
I agree, and that's how I had it originally, but we use that module in `TestingTrinoServer` transitively via `ServerMainModule`, so this seems like the cleanest way. It's often good practice to split up Guice modules so that testing code (or different uses cases) can provide alternate implementations, which is what we are doing here.

Reply:
An alternative is to set `discovery-server.enabled=false` in `TestingTrinoServer`, but that means using a deprecated property we intend to remove, so it would be short lived. Also, that would prevent using the `@AssertTrue` validation that I previously mentioned, which might be a good next step towards removing it.